### PR TITLE
Feature/items ordination

### PR DESCRIPTION
**Pull request by marcosewbank**
#### What problem is this solving?

This new feature will allow to sort items at `orderForm` by name or date of addition using the new `itemsOrdination` mutation.

#### How should this be manually tested?

1. Go to this [Workspace](https://ecom5991--carrefourbrfood.myvtex.com/) and add some products to the cart

2. Run this [mutation](https://ecom5991--carrefourbrfood.myvtex.com/_v/private/vtex.checkout-graphql@0.44.1/graphiql/v1?operationName=itemsOrdination&query=mutation%20itemsOrdination(%24ascending%3A%20Boolean!%2C%20%24criteria%3A%20String!)%20%7B%0A%20%09itemsOrdination(ascending%3A%20%24ascending%2C%20criteria%3A%20%24criteria)%20%7B%0A%20%09%09id%0A%20%20%20%20items%7B%0A%20%20%20%20%20%20name%0A%20%20%20%20%20%20skuName%0A%20%20%20%20%7D%0A%09%7D%0A%7D%20&variables=%7B%0A%20%20%22ascending%22%3A%20true%2C%0A%20%20%22criteria%22%3A%20%22add_time%22%0A%7D)
3. Then reload the page.

Now the products are reordered :smiley: 

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️| New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

**Comment by lucasecdb** [comment]
@marcosewbank can you remove the commit 0ad4bd5 and rebase against master instead?

File: graphql/schema.graphql [hunk 1]
```diff
@@ -81,4 +81,8 @@ type Mutation {
   setManualPrice(orderFormId: ID, input: ManualPriceInput!): OrderForm!
     @withOrderFormId
     @cacheControl(scope: PRIVATE)
+
+  itemsOrdination(ascending: Boolean!, criteria: String!): OrderForm!
```
Comment (lucasecdb) [inline graphql/schema.graphql:85]: add the orderFormId argument here as well
```suggestion
  itemsOrdination(orderFormId: ID, ascending: Boolean!, criteria: String!): OrderForm!
```

File: node/resolvers/orderForm.ts [hunk 2]
```diff
@@ -259,4 +259,24 @@ export const mutations = {
 
     return orderFormWithPayments
   },
+
+  itemsOrdination: async (
+    _: unknown,
+    { ascending, criteria }: ItemsOrdinationArgs,
```
Comment (lucasecdb) [inline node/resolvers/orderForm.ts:265]:
```suggestion
    args: ItemsOrdinationArgs & OrderFormIdArgs,
```

File: node/resolvers/orderForm.ts [hunk 3]
```diff
@@ -259,4 +259,24 @@ export const mutations = {
...
+    ctx: Context
+    ): Promise<CheckoutOrderForm> => {
+
+    const {
+      clients: { checkout },
+      vtex: { orderFormId },
+    } = ctx
```
Comment (lucasecdb) [inline node/resolvers/orderForm.ts:272]: You should use the orderFormId from the mutation input, with a default of the id from cookies

```suggestion
    const { clients: { checkout }, vtex } = ctx
    const {
      ascending,
      criteria,
      orderFormId = vtex.orderFormId,
    } = args
```

File: graphql/schema.graphql (hunk 1 above)
Comment (lucasecdb) [inline graphql/schema.graphql:85]: also, I think `updateItemsOrdination` is a better name for a mutation, but could be something like `setItemsOrdination` too
```suggestion
  updateItemsOrdination(ascending: Boolean!, criteria: String!): OrderForm!
```

**Comment by marcosewbank** [comment]
@lucasecdb i made the requested changes, thank you for the amazing tips!! :smiley: could you please review again to see if is everithing okay?

File: graphql/schema.graphql [hunk 4]
```diff
@@ -81,4 +81,8 @@ type Mutation {
   setManualPrice(orderFormId: ID, input: ManualPriceInput!): OrderForm!
     @withOrderFormId
     @cacheControl(scope: PRIVATE)
+
+  updateItemsOrdination(orderFormId: ID, ascending: Boolean!, criteria: String!): OrderForm!
```
Comment (lucasecdb) [inline graphql/schema.graphql:85]: can you create an enum for the criteria, instead of using a plain string?

File: node/typings/global.d.ts [hunk 5]
```diff
@@ -532,4 +532,9 @@ declare global {
   interface PaymentDataInput {
     payments: PaymentInput[]
   }
+
+  interface ItemsOrdinationArgs {
+    ascending: boolean,
```
Comment (lucasecdb) [inline node/typings/global.d.ts:537]: nitpick
```suggestion
    ascending: boolean
```

**Review by lucasecdb — DISMISSED** [review]
LGTM

**Comment by lucasecdb** [comment]
@marcosewbank you still need to drop the merge commits (0ad4bd5 and 2368659) and [rebase](https://git-scm.com/docs/git-rebase) against master, you can use `git push --force` after the rebase to avoid creating another merge commit

**Comment by marcosewbank** [comment]
I closed the pull request because my forked repo wasn't updated when i started the feature, due to that a lot of commits was added unecessearly, please evaluate here: https://github.com/vtex-apps/checkout-graphql/pull/102

**Pull request closed without merging**